src/io/config.cpp

```diff
@@ -215,12 +215,14 @@ void Config::CheckParamConflict() {
     }
   }
   for (std::string metric_type : metric) {
+    bool metric_custom = metric_type == std::string("none") || metric_type == std::string("null") || metric_type == std::string("custom");
```

Comment:
I think this should be consistent with `aliases` here:
https://github.com/Microsoft/LightGBM/blob/1e53445e85fe7d9597a274e4ddc1b1af190d0a64/include/LightGBM/config.h#L662